src/Nethermind/Nethermind.Core/Caching/MemCountingCache.cs

```diff
@@ -3,199 +3,250 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using Nethermind.Core.Crypto;
-using Nethermind.Core.Extensions;
 
-namespace Nethermind.Core.Caching
+namespace Nethermind.Core.Caching;
+
+public sealed class MemCountingCache : ICache<KeccakKey, byte[]>
 {
-    /// <summary>
-    /// https://stackoverflow.com/questions/754233/is-it-there-any-lru-implementation-of-idictionary
-    /// </summary>
-    public class MemCountingCache : ICache<Keccak, byte[]>
-    {
-        private readonly int _maxCapacity;
-        private readonly Dictionary<Keccak, LinkedListNode<LruCacheItem>> _cacheMap;
-        private readonly LinkedList<LruCacheItem> _lruList;
+    private readonly int _maxCapacity;
+    private readonly Dictionary<KeccakKey, LinkedListNode<LruCacheItem>> _cacheMap;
+    private LinkedListNode<LruCacheItem>? _first;
 
-        private const int PreInitMemorySize =
-            48 /* LinkedList */ +
-            80 /* Dictionary */ +
-            MemorySizes.SmallObjectOverhead +
-            8 /* sizeof(int) aligned */;
+    private const int PreInitMemorySize =
+        48 /* LinkedList */ +
```

Comment:
we don't have LinkedList now, is it accurate?

Reply:
Updated